tests/test_core.py

```diff
@@ -25,31 +25,42 @@ class test_PairMatrix(unittest.TestCase):
     def test_init(self):
         """Test construction with different list types."""
 
-        pair_types = ('A','B')
-        pair_dict = {('A','B'):{}, ('B','B'):{}, ('A','A'):{}}
+        types_real = ('A','B')
+        pairs_real  = (('A','B'), ('B','B'), ('A','A'))
 
         #test construction with tuple input
         m = relentless.core.PairMatrix(types=('A','B'))
-        self.assertEqual(m.types, pair_types)
-        self.assertEqual(m._data, pair_dict)
+        self.assertEqual(m.types, types_real)
+        self.assertCountEqual(m.pairs, pairs_real)
 
         #test construction with list input
         m = relentless.core.PairMatrix(types=['A','B'])
-        self.assertEqual(m.types, pair_types)
-        self.assertEqual(m._data, pair_dict)
+        self.assertEqual(m.types, types_real)
+        self.assertCountEqual(m.pairs, pairs_real)
 
-        #test construction with numpy array input
-        m = relentless.core.PairMatrix(types=np.array(['A','B']))
-        self.assertEqual(m.types, pair_types)
-        self.assertEqual(m._data, pair_dict)
-
-        pair_types = ('A',)
-        pair_dict = {('A','A'):{}}
+        types_real = ('A',)
+        pairs_real = (('A','A'),)
 
         #test construction with single type tuple
         m = relentless.core.PairMatrix(types=('A',))
-        self.assertEqual(m.types, pair_types)
-        self.assertEqual(m._data, pair_dict)
+        self.assertEqual(m.types, types_real)
+        self.assertCountEqual(m.pairs, pairs_real)
+
+        types_real = (1,2)
+        pairs_real = ((1,2), (2,2), (1,1))
+
+        #test construction with int type input
+        m = relentless.core.PairMatrix(types=(1,2))
+        self.assertEqual(m.types, types_real)
+        self.assertCountEqual(m.pairs, pairs_real)
+
+        types_real = ('1',2)
+        pairs_real = (('1',2), (2,2), ('1','1'))
+
+        #test construction with mixed type input
+        m = relentless.core.PairMatrix(types=('1',2))
+        self.assertEqual(m.types, types_real)
+        self.assertCountEqual(m.pairs, pairs_real)
```

Comment:
On thinking further, let's make this an error to mix two types that don't have comparators implemented. Just check that an error is raised.